pkg/auth/providers/common/usermanager.go

```diff
@@ -149,6 +149,14 @@ func (m *userManager) SetPrincipalOnCurrentUserByUserID(userID string, principal
 		return nil, err
 	}
 
+	// ensure this principal is unique to this user
+	if conflict, err := m.GetUserByPrincipalID(principal.Name); err != nil {
+		return nil, err
+	} else if conflict != nil && conflict.UID != user.UID {
+		logrus.Errorf("Cannot set principal \"%s\" on user \"%s\". Principal already in use on user \"%s\".", principal.Name, user.UID, conflict.UID)
+		return nil, errors.New("cannot set principal on user that is already in use")
```

Comment:
Is the goal to inform the user with this error and then they have to lookup the error in the logs to find what is causing it? Is this for security reasons?

Reply:
I did separate error messages because this error is returned on a callback with the external provider.  I want to limit external exposures of user information as much as possible.  The user will need to check the logs to debug this anyway.  The log message should have enough information for them to take action.